src/main/io/serial_cli.c

```diff
@@ -588,29 +588,42 @@ static void cliRxFail(char *cmdline)
 
             uint16_t value;
             rxFailsafeChannelMode_e mode = channelFailsafeConfiguration->mode;
-            bool requireValue = channelFailsafeConfiguration->mode == RX_FAILSAFE_MODE_SET;
+            bool requireValue = ((mode == RX_FAILSAFE_MODE_AUTO_OR_SET) && (channel >= NON_AUX_CHANNEL_COUNT));
+            bool parseError = true;
 
             // TODO optimize to use rxFailsafeModes - less logic.
             ptr = strchr(ptr, ' ');
             if (ptr) {
                 switch (*(++ptr)) {
-                    case 'a':
-                        mode = RX_FAILSAFE_MODE_AUTO;
-                        break;
-
                     case 'h':
                         mode = RX_FAILSAFE_MODE_HOLD;
+                        parseError = false;
+                        break;
+
+                    case 'a':
+                        if (channel < NON_AUX_CHANNEL_COUNT) {
+                            mode = RX_FAILSAFE_MODE_AUTO_OR_SET;
+                            parseError = false;
+                        }
                         break;
 
                     case 's':
-                        mode = RX_FAILSAFE_MODE_SET;
+                        if (channel >= NON_AUX_CHANNEL_COUNT) {
+                            mode = RX_FAILSAFE_MODE_AUTO_OR_SET;
+                            parseError = false;
+                        }
                         break;
+
                     default:
-                        cliShowParseError();
-                        return;
+                        break;
+                }
+
```

Comment:
instead of checking the channel value twice, use the assigned `mode` value and the `isFlightChannelVariable`

then the `TODO` on line 593 can still be done and the entire switch statement can be replaced with a lookup.